contracts/PeeranhaToken.sol

```diff
@@ -65,42 +92,93 @@ contract PeeranhaToken is ERC20Upgradeable, ERC20PausableUpgradeable, ERC20Cappe
-    _mint(user, userReward * FRACTION);
+    _mint(user, userReward);
+  }
+
+  function setBoost(address user, uint256 stakeTokens) external {
+    require(stakeTokens <= balanceOf(user), "Boost amount exceeds balance");
+    uint16 nextPeriod = RewardLib.getPeriod(CommonLib.getTimestamp()) + 2;
+    TokenLib.UserBoost storage userBoost = weekUserBoost.weekUserBoost[user];
+
+    TokenLib.CountBoost storage countBoost = boostContainer.userBoostCount[nextPeriod];
+    if (boostContainer.updateBoostInPeriod.length != 0 && boostContainer.updateBoostInPeriod[boostContainer.updateBoostInPeriod.length - 1] != nextPeriod) {
+      uint16 previousPeriod = boostContainer.updateBoostInPeriod[boostContainer.updateBoostInPeriod.length - 1];
+      countBoost.allStake = boostContainer.userBoostCount[previousPeriod].allStake;
+      countBoost.countStaking = boostContainer.userBoostCount[previousPeriod].countStaking;
+      boostContainer.updateBoostInPeriod.push(nextPeriod);
+    } else if (boostContainer.updateBoostInPeriod.length == 0) {
+      boostContainer.updateBoostInPeriod.push(nextPeriod);
+    }
+
+    if (userBoost.updateUserBoostInPeriod.length != 0) {
+      uint16 lastBoostPeriod = userBoost.updateUserBoostInPeriod[userBoost.updateUserBoostInPeriod.length - 1];
+      if (lastBoostPeriod != nextPeriod) {
+        userBoost.updateUserBoostInPeriod.push(nextPeriod);
+      }
+
+      if (userBoost.stakedTokens[lastBoostPeriod] > stakeTokens) {
+        countBoost.allStake += userBoost.stakedTokens[lastBoostPeriod] - stakeTokens;
+        if (stakeTokens == 0) countBoost.countStaking--;
+      } else {
+        countBoost.allStake += stakeTokens - userBoost.stakedTokens[lastBoostPeriod];
+        if (userBoost.stakedTokens[lastBoostPeriod] == 0) countBoost.countStaking++;
+      }
+      userBoost.stakedTokens[nextPeriod] = stakeTokens;
+
+    } else {
+      userBoost.updateUserBoostInPeriod.push(nextPeriod);
+      userBoost.stakedTokens[nextPeriod] = stakeTokens;
+
+      countBoost.allStake += stakeTokens;
+      countBoost.countStaking++;
+    }
   }
 
   function getPool(RewardLib.WeekReward memory weekReward, uint16 period) private view returns(uint256) {
     uint256 poolToken = poolTokens[period];
     if (poolToken == 0) {
-      poolToken = reduceRewards(REWARD_WEEK, period);
+      poolToken = reduceRewards(REWARD_WEEK * FRACTION, period);                // max_pool
       if (weekReward.activeUsersInPeriod.length <= ACTIVE_USERS_IN_PERIOD) {
-        uint256 userRewardWeek = weekReward.activeUsersInPeriod.length * 1000 * FRACTION;
+        uint256 userRewardWeek = weekReward.activeUsersInPeriod.length * USER_REWARD * FRACTION;  // < 1000
         poolToken = CommonLib.minUint256(poolToken, userRewardWeek);
       }
-      poolToken = CommonLib.minUint256(poolToken, uint256(weekReward.rating) * TokenLib.getRewardCoefficient());
+      poolToken = CommonLib.minUint256(poolToken, weekReward.tokens);   // earned 
```

Comment:
What is `weekReward.tokens`??

Reply:
I don't think `CommonLib.minUint256` is needed here. `weekReward.tokens` is not tokens but total of `user_rating * boost`.